test/features/site_channel_verification_test.rb

```diff
@@ -31,6 +31,32 @@ class SiteChannelVerificationTest < Capybara::Rails::TestCase
     refute_content "Cancel"
   end
 
+  test "When bad ssl happens" do
+    Rails.application.secrets[:host_inspector_offline] = false
+
+    publisher = publishers(:default)
+    sign_in publisher
+
+    visit new_site_channel_path
+    assert_content "Cancel"
+
+    fill_in "channel_details_attributes_brave_publisher_id_unnormalized", with: "https://self-signed.badssl.com/"
+
+    click_button("Continue")
+    channel = publisher.channels.order("created_at").last
+    assert_current_path verification_choose_method_site_channel_path(channel)
+    assert_content "Cancel"
+
+    click_link "I'll use a trusted file"
+    assert_current_path verification_public_file_site_channel_path(channel)
+    assert_content "Choose Different Verification Method"
+    refute_content "Cancel"
+
+    assert_content "The following error was encountered: SSL_connect returned=1 errno=0 state=error: certificate verify failed"
+
+    Rails.application.secrets[:host_inspector_offline] = true
```

Comment:
(minor) This won't apply if we stub the responses, but setting the `Rails.application.secrets[:host_inspector_offline] = true` assumes that it was previously set to true, which may not be the case (now or in the future).  We've been using this convention to change envars for specific tests:

https://github.com/brave-intl/publishers/blob/d6f02765c389d99e62026edf9865c94b0aad7680/test/controllers/admin/payout_reports_controller_test.rb#L8-L14

Then you just declare what value you want the envar to be at the top of the test.

Reply:
Ah! Nice catch. I wouldn't have thought about that since I'm used to RSpec isolating all my tests. I'm going to switch over to mocking the request so we don't have to worry about this.

Reply:
Actually, I did need to set this so the request is actually mocked - https://github.com/brave-intl/publishers/pull/1345/commits/19075a783167f44b10d0b0d74477b2098a82f08d